test/e2e/page-objects/flows/user-operations.flow.ts

```diff
@@ -0,0 +1,189 @@
+  await snapInstall.checkPageIsLoaded();
+  await snapInstall.clickConfirmButton();
+  await snapInstall.clickOkButton();
+}
+
+/**
+ * Flow for setting up the ERC-4337 snap with configuration and creating an account.
+ *
+ * @param driver - The WebDriver instance
+ * @param config - Configuration object for the snap
+ * @param config.bundlerUrl - The bundler URL
+ * @param config.entrypoint - The entry point address
+ * @param config.simpleAccountFactory - The simple account factory address
+ * @param config.paymaster - Optional paymaster address
+ * @param config.paymasterSK - Optional paymaster secret key
+ * @param accountConfig - Account creation configuration
+ * @param accountConfig.privateKey - The private key for the account
+ * @param accountConfig.salt - The salt for account creation
+ */
+export async function setupERC4337SnapAccount(
+  driver: Driver,
+  config: {
+    bundlerUrl: string;
+    entrypoint: string;
+    simpleAccountFactory: string;
+    paymaster?: string;
+    paymasterSK?: string;
+  },
+  accountConfig: {
+    privateKey: string;
+    salt: string;
+  },
+): Promise<void> {
+  console.log('Setting up ERC-4337 snap account');
+
+  const erc4337SnapPage = new ERC4337SnapPage(driver);
+
+  // Configure the snap
+  await erc4337SnapPage.setSnapConfiguration(config);
+
+  // Create the snap account
+  await erc4337SnapPage.createSnapAccount(
+    accountConfig.privateKey,
+    accountConfig.salt,
+  );
+}
+
+/**
+ * Flow for connecting a test dapp to the ERC-4337 snap account.
+ *
+ * @param driver - The WebDriver instance
+ * @param accountAddress - The public address of the ERC-4337 account
+ */
+export async function connectTestDappToSnapAccount(
+  driver: Driver,
+  accountAddress: string,
+): Promise<void> {
+  console.log('Connecting test dapp to snap account');
+
+  const testDapp = new TestDapp(driver);
+  await testDapp.openTestDappPage();
+  await testDapp.connectAccount({ publicAddress: accountAddress });
+}
+
+/**
+ * Flow for confirming a transaction in the notification window.
+ *
+ * @param driver - The WebDriver instance
+ */
+export async function confirmTransaction(driver: Driver): Promise<void> {
+  console.log('Confirming transaction');
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
+  await driver.clickElement({ text: 'Confirm' });
+}
+
+/**
+ * Flow for opening a confirmed transaction from the activity list.
+ *
+ * @param driver - The WebDriver instance
+ */
+export async function openConfirmedTransaction(driver: Driver): Promise<void> {
+  console.log('Opening confirmed transaction from activity list');
+
+  await driver.switchToWindowWithTitle(WINDOW_TITLES.ExtensionInFullScreenView);
+
+  const homePage = new HomePage(driver);
+  await homePage.goToActivityList();
+
+  // Wait for confirmed transaction to appear and click on it
+  await driver.clickElement({
+    css: '[data-testid="activity-list-item"]',
+    text: 'Confirmed',
+  });
+}
```

Comment:
ℹ️  We could either remove this and use the Activity List class directly in the spec, or add a flow if you consider it, but this should live in a separate file, not under `user-operations` as those are unrelated